### ISSUE TITLE
compilation issue

### ISSUE THREAD

**Issue by giaf** [mkatliar/blasfeo_benchmark#3]
```
g++ -std=c++17 -g -O2 -DNDEBUG \
	-I./benchmark/include \
	-L \
	-I./benchmark/include \
	-L./benchmark/build/src \
	-I./blasfeo/include \
	-L./blasfeo/lib \
	Gemm.cpp Main.cpp -lbenchmark -lblasfeo -lpthread -lmkl_rt -o bench
Gemm.cpp:36:16: error: expected ‘{’ before ‘::’ token
 namespace tmpc :: benchmark
                ^
Gemm.cpp:36:19: error: ‘benchmark’ in namespace ‘::’ does not name a type
 namespace tmpc :: benchmark
                   ^
Gemm.cpp:296:1: error: expected ‘}’ at end of input
 }
 ^
Makefile:19: recipe for target 'bench' failed
make: *** [bench] Error 1
```

compiler
```
g++ --version
g++ (Ubuntu 5.5.0-12ubuntu1~16.04) 5.5.0 20171010
```

**Comment by mkatliar** [comment]
Too old g++ which does not understand nested namespace syntax. Just remove the entire namespace (or I will do it when I reach a PC).

**Comment by tmmsartor** [comment]
I am not sure for which reason this is not supported in c++ <=14
but if you remove `tmpc::` from the definition of the namespace you will solve this first issue.

But you will get another issue because [std::aligned_alloc](https://en.cppreference.com/w/cpp/memory/c/aligned_alloc) has been added in c++17, while [Gcc 5.5 supports only c++14](https://gcc.gnu.org/projects/cxx-status.html) and previous standard.

One possible solution is to install gcc7, to do that in ubuntu 16.04 you can see [here](https://askubuntu.com/a/887791/393204).

**Comment by mkatliar** [comment]
Fixed in bbc820bf363b71909bad2622575c17c355df7677
